internal/upload/koji/koji.go

```diff
@@ -323,27 +322,29 @@ func (k *Koji) uploadChunk(chunk []byte, filepath, filename string, offset uint6
 			if v, ok := err.(*url.Error); ok {
 				if _, ok := v.Err.(x509.UnknownAuthorityError); ok {
 					// retry if it's a timeout
-					return strings.Contains(strings.ToLower(v.Error()), "timeout"), v
+					shouldRetry = strings.Contains(strings.ToLower(v.Error()), "timeout")
+					retErr = v
 				}
 			}
 		}
 
+		if shouldRetry {
+			retries++
+		}
+
 		return shouldRetry, retErr
 	}
 
-	client := rh.Client{
-		HTTPClient: &http.Client{
-			Transport: k.transport,
-		},
-		CheckRetry: countingCheckRetry,
-		Logger:     rh.LeveledLogger(&LeveledLogrus{logrus.StandardLogger()}),
+	client := rh.NewClient()
+
+	client.HTTPClient = &http.Client{
+		Transport: k.transport,
 	}
+	client.CheckRetry = countingCheckRetry
+	client.Logger = rh.LeveledLogger(&LeveledLogrus{logrus.StandardLogger()})
 
 	respData, err := client.Post(u.String(), "application/octet-stream", bytes.NewBuffer(chunk))
 
-	// don't count the first call since it's not a retry
```

Comment:
The function is not called for the original HTTP call, only for retries, so no need to decrement.